### PR TITLE
added python solution for cses-1095.mdx

### PR DESCRIPTION
**Pull request by praneethNellutla**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [X] I have tested my code.
- [X] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [X] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [ ] I have linked this PR to any issues that it closes.

### CODE REVIEW

File: solutions/gold/cses-1095.mdx [hunk 1]
```diff
@@ -56,4 +56,31 @@ int main() {
...
+import sys
+
+input = sys.stdin.readline
+takeInt = lambda: int(input())
+takeInts = lambda: map(int, input().split())
+takeList = lambda: list(map(int, input().split()))
+mod = int(1e9) + 7
+
+
+def power(base, exponent, mod):
```
Comment (SansPapyrus683) [inline solutions/gold/cses-1095.mdx:71]: python has builtin modular exponentiation, use that please

Comment (praneethNellutla) [reply solutions/gold/cses-1095.mdx:71]: yeah but i want to show the process of binary exponentiation to perform in python so i did it. do i need to change it now

Comment (SansPapyrus683) [reply solutions/gold/cses-1095.mdx:71]: yeah

Comment (envyaims) [reply solutions/gold/cses-1095.mdx:71]: the whole point of the problem is to use modular exponentiation. I think he should keep it.

File: solutions/gold/cses-1095.mdx [hunk 2]
```diff
@@ -56,4 +56,31 @@ int main() {
...
 </CPPSection>
+
+<PySection>
+```py
+import sys
+
+input = sys.stdin.readline
+takeInt = lambda: int(input())
+takeInts = lambda: map(int, input().split())
+takeList = lambda: list(map(int, input().split()))
```
Comment (envyaims) [inline solutions/gold/cses-1095.mdx:67]: remove these, you only call them once anyways.

**Review by envyaims — CHANGES_REQUESTED** [review]

File: solutions/gold/cses-1095.mdx (hunk 1 above)
Comment (SansPapyrus683) [reply solutions/gold/cses-1095.mdx:71]: look here: https://usaco.guide/gold/modular?lang=py#solution---exponentiation

if people wanna look at the method behind it they can look up how the `pow` works

File: solutions/gold/cses-1095.mdx (hunk 2 above)
Comment (praneethNellutla) [reply solutions/gold/cses-1095.mdx:67]: ok i will change them. please give me the clarity do i use the inbuilt pow function or my custom function

Comment (SansPapyrus683) [reply solutions/gold/cses-1095.mdx:67]: use the builtin

Comment (envyaims) [reply solutions/gold/cses-1095.mdx:67]: builtin might tle

Comment (envyaims) [reply solutions/gold/cses-1095.mdx:67]: can't we include both as two solutions?

Comment (SansPapyrus683) [reply solutions/gold/cses-1095.mdx:67]: in the module, perhaps, but this solution should definitely use the builtin

**Review by SansPapyrus683 — APPROVED** [review]

File: solutions/gold/cses-1095.mdx [hunk 3]
```diff
@@ -39,14 +39,10 @@ ll exp(ll x, ll n, ll m) {
...
-	for (int i = 0; i < t; i++) {
-		ll a;
-		ll b;
-		ll c;
-
-		cin >> a >> b >> c;
+	int test_num;
+	cin >> test_num;
+	for (int t = 0; t < test_num; t++) {
+		ll a, b, c cin >> a >> b >> c;
```
Comment (ryanchou-dev) [inline solutions/gold/cses-1095.mdx:45]: add semicolon

**Review by ryanchou-dev — CHANGES_REQUESTED** [review]

**Review by ryanchou-dev — APPROVED** [review]

**Pull request merged**